lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -151,68 +198,121 @@
     - device_name: /dev/sdb
       ephemeral: ephemeral0
 
+# create a launch configuration using a running instance id as a basis
+
+- ec2_lc:
+    name: special
+    instance_id: i-00a48b207ec59e948
+    key_name: default
+    security_groups: ['launch-wizard-2' ]
+    volumes:
+    - device_name: /dev/sda1
+      volume_size: 120
+      device_type: io1
+      iops: 3000
+      delete_on_termination: true
+
 '''
-import traceback
 
-from ansible.module_utils.basic import *
-from ansible.module_utils.ec2 import ec2_argument_spec, ec2_connect, connect_to_aws, \
-    get_ec2_security_group_ids_from_names, get_aws_connection_info, AnsibleAWSError
+import traceback
+from ansible.module_utils.ec2 import (get_aws_connection_info,
+                                      ec2_argument_spec,
+                                      ec2_connect,
+                                      camel_dict_to_snake_dict,
+                                      get_ec2_security_group_ids_from_names,
+                                      boto3_conn, snake_dict_to_camel_dict,
+                                      HAS_BOTO3)
+from ansible.module_utils.basic import AnsibleModule
 
 try:
-    from boto.ec2.blockdevicemapping import BlockDeviceType, BlockDeviceMapping
-    import boto.ec2.autoscale
-    from boto.ec2.autoscale import LaunchConfiguration
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
-except ImportError:
-    HAS_BOTO = False
-
-
-def create_block_device(module, volume):
-    # Not aware of a way to determine this programatically
-    # http://aws.amazon.com/about-aws/whats-new/2013/10/09/ebs-provisioned-iops-maximum-iops-gb-ratio-increased-to-30-1/
+    import botocore
+except Exception:
+    pass
+
+
+def create_block_device_meta(module, volume):
     MAX_IOPS_TO_SIZE_RATIO = 30
     if 'snapshot' not in volume and 'ephemeral' not in volume:
         if 'volume_size' not in volume:
-            module.fail_json(msg='Size must be specified when creating a new volume or modifying the root volume')
+            module.fail_json(msg='Size must be specified when creating a new ' +
+                             'volume or modifying the root volume')
     if 'snapshot' in volume:
-        if 'device_type' in volume and volume.get('device_type') == 'io1' and 'iops' not in volume:
+        if 'device_type' in volume and volume.get(
+                'device_type') == 'io1' and 'iops' not in volume:
```

Comment:
done - thanks for this. I always thought it was 79 characters for some reason. I'll apply the same 160 char limit to #24292 while I think of it to make it more readable.